dashboard-api/internal/study.go

```diff
@@ -0,0 +1,49 @@
+package studiesmanager
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+type StudyRepository interface {
+	GetStudyBySlug(ctx context.Context, slug string) (Study, error)
+}
+
+var ErrStudyNotFound = errors.New("Study not found")
+
+//go:generate mockery --case=snake --outpkg=storagemocks --output=platform/storage/storagemocks --name=StudyRepository
+
+type Study struct {
+	id        string
+	name      string
+	slug      string
+	createdAt time.Time
+}
+
+func NewStudy(id, name, slug string, createdAt time.Time) Study {
+	return Study{
+		id:        id,
+		name:      name,
+		slug:      slug,
+		createdAt: createdAt,
+	}
+}
+
+func (s Study) ID() string {
```

Comment:
Why have a getter for this, why isn't it just public? Do you really think anyone's going to overwrite it? That's a very Java-esque paradigm, I don't see the advantage.

Reply:
You guessed it right, I used the getters to achieve immutability.

But I agree with you, looking at it again, it's overkill and the problem they may solve, is non existent for now.

I'll update the code to get rid of them. 👍